qiskit/assembler/disassemble.py

```diff
@@ -165,16 +179,12 @@ def _experiments_to_circuits(qobj):
                     clbits.append(creg_dict[clbit_label[0]][clbit_label[1]])
             except Exception:  # pylint: disable=broad-except
                 pass
-            if hasattr(circuit, name):
+            # TODO remove the check that name is not in the deprecated circuit methods
+            # once the methods have been removed
+            if hasattr(circuit, name) and name not in _DEPRECATED_CIRCUIT_METHODS:
```

Comment:
Strictly you might want `_DEPRECATED_CIRCUIT_METHODS` to be some sort of `set` if you want to do many lookups into it.